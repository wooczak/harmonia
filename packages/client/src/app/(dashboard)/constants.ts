export const Routes = {
  HOMEPAGE: "/",
  PATIENTS: "/patients",
  CALENDAR: "/calendar",
  NOTES: "/notes",
  ACCOUNT: "/acccount",
  SETTINGS: "/settings"
};

export const NavigationTabs = [
  'Pacjenci/Pacjentki',
  'Kalendarz',
  'Notatki',
  'Forum'
];