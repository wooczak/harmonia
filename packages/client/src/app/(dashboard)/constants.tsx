import { CgProfile } from 'react-icons/cg'
import { FiSettings } from 'react-icons/fi'
import { AiOutlineHome } from 'react-icons/ai';

export const Heading = {
  AppName: 'Harmonia'
}

export const Routes = {
  HOMEPAGE: '/',
  PATIENTS: '/patients',
  CALENDAR: '/calendar',
  NOTES: '/notes',
  CHAT: '/chat',
  ACCOUNT: '/account',
  SETTINGS: '/settings'
}

export const NavigationTabs = {
  [Routes.HOMEPAGE]: <AiOutlineHome size={20} />,
  [Routes.PATIENTS]: 'Pacjenci/Pacjentki',
  [Routes.CALENDAR]: 'Kalendarz',
  [Routes.NOTES]: 'Notatki',
  [Routes.CHAT]: 'Czat',
  [Routes.ACCOUNT]: <CgProfile />,
  [Routes.SETTINGS]: <FiSettings />
}
