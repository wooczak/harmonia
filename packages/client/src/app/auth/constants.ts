export const RegExp = {
  Email: /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/gi,
  Password: /^(?=.*[A-Z])(?=.*[!:.?;,])[\w!:.?;,]{8,}$/
}

export const Error = {
  Email: "To pole musi zawierać poprawną formę adresu e-mail.",
  Password: "Hasło musi zawierać co najmniej 8 znaków, co najmniej jedną wielką literę oraz co najmniej jeden znak specjalny."
}