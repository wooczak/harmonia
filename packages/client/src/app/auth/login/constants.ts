export const LoginPageConstants = {
  HEADING: 'Welcome back to Harmonia',
  SIGN_IN_TEXT: 'Sign in to your account below.',
  FORGOT_PASSWORD: 'Forgot your password?',
  NO_ACCOUNT: `Don't have an account?`,
  REGISTER_NOW: 'Register now.',
  REMEMBER_ME: 'Remember me'
}
