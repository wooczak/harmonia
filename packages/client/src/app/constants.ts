export const Routes = {
  HOMEPAGE: "/",
  PATIENTS: "/patients"
};

export const NavigationTabs = [
  'Pacjenci/Pacjentki',
  'Kalendarz',
  'Notatki',
  'Forum'
];