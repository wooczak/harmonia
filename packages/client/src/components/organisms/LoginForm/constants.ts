export const LoginPageConstants = {
  HEADING_START: `Welcome back`,
  HEADING_END: 'to Harmonia',
  SIGN_IN_TEXT: 'Sign in to your account below.',
  FORGOT_PASSWORD: 'Forgot your password?',
  NO_ACCOUNT: `Don't have an account?`,
  REGISTER_NOW: 'Register now.',
  REMEMBER_ME: 'Remember me',
  LOG_IN: 'Log in'
}

export const InputPlaceholder = {
  E_MAIL: "Enter your e-mail here",
  PASSWORD: "Type your password here"
}
