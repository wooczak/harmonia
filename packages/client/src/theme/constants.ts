export const Color = {
  BLACK: '#202020',
  YELLOW: '#F2C744',
  LIGHT_GREY: '#F2F0EE'
}