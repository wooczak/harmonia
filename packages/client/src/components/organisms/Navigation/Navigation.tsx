'use client'

import Link from 'next/link'
import { usePathname } from 'next/navigation'
import React, { ReactNode, useCallback } from 'react'
import { AiOutlineHome } from 'react-icons/ai'
import { renderTabs } from './helpers'
import { Routes } from 'src/app/(dashboard)/constants'
import Title from 'src/components/atoms/Title'

export type NavTabs = { [x: string]: string | ReactNode }

type TNavigationProps = {
  tabs: NavTabs
  titleContent: string
}

export default function Navigation({ tabs, titleContent }: TNavigationProps) {
  const pathname = usePathname()

  const renderNavTabs = useCallback(
    function (tabs: NavTabs) {
      return renderTabs(tabs)
    },
    [tabs]
  )

  const { HOMEPAGE } = Routes

  return (
    <div className="flex h-[5rem] border-b-2 border-yellow-100 mb-5">
      <Title type="Primary" content={titleContent} href={HOMEPAGE} customStyling="leading-[5rem]"></Title>
      <nav className="flex items-center grow">
        <ul className="flex gap-10 w-full justify-end items-center">
          {pathname !== HOMEPAGE && (
            <Link href={HOMEPAGE}>
              <AiOutlineHome size={20} />
            </Link>
          )}
          {renderNavTabs(tabs)}
        </ul>
      </nav>
    </div>
  )
}
