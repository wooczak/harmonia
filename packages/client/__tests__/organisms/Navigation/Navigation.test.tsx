import React from 'react'
import renderer from 'react-test-renderer'
import Navigation from '../../../src/components/organisms/Navigation/Navigation'
import { Heading, NavigationTabs } from '../../../src/app/(dashboard)/constants'

describe('WHEN the Navigation organism is rendered', function () {
  it('THEN should render correctly', function () {
    const tree = renderer
      .create(<Navigation isUserAtHomePage={false} tabs={NavigationTabs} titleContent={Heading.AppName} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
