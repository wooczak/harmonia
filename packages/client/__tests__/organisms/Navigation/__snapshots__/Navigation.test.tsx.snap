// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHEN the Navigation organism is rendered THEN should render correctly 1`] = `
<nav
  className="flex items-center grow"
>
  <h1
    className="font-black text-4xl text-black  leading-[5rem] hover:text-yellow-300 duration-200 ease-in-out"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Harmonia
    </a>
  </h1>
  <ul
    className="flex gap-10 w-full justify-end items-center"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        href="/patients"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Pacjenci/Pacjentki
      </a>
    </li>
    <li>
      <a
        href="/calendar"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Kalendarz
      </a>
    </li>
    <li>
      <a
        href="/notes"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Notatki
      </a>
    </li>
    <li>
      <a
        href="/chat"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Czat
      </a>
    </li>
    <li>
      <a
        href="/acccount"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
            fill="currentColor"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        href="/settings"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;
